### PR TITLE
Fix publishing Custom Analytics Events Data

### PR DESCRIPTION
**Pull request by YasasRangika**
## Purpose
To fix missing properties map when publishing custom properties to on-prem analytics.

Related PR: https://github.com/wso2/apim-analytics-publisher/pull/94

### CODE REVIEW

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/elk/ELKCounterMetric.java [hunk 1]
```diff
@@ -58,11 +61,26 @@ public String getName() {
...
+                        GenericInputValidator.getInstance().getEventProperties(MetricSchema.ELK_RESPONSE));
+            case ERROR:
+                return new ELKMetricEventBuilder(
+                        GenericInputValidator.getInstance().getEventProperties(MetricSchema.ELK_ERROR));
+            case CHOREO_RESPONSE:
+                return new ELKMetricEventBuilder(
+                        GenericInputValidator.getInstance().getEventProperties(MetricSchema.CHOREO_RESPONSE));
+            case CHOREO_ERROR:
+                return new ELKMetricEventBuilder(
+                        GenericInputValidator.getInstance().getEventProperties(MetricSchema.CHOREO_ERROR));
```
Comment (ruks) [inline component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/elk/ELKCounterMetric.java:83]: ELK report will not used my choreo CC. So we can remove this

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/cloud/DefaultResponseMetricEventBuilder.java [hunk 2]
```diff
@@ -119,10 +119,10 @@ private void setUserAgentProperties(String userAgentHeader) {
...
 
     private void copyDefaultPropertiesToRootLevel(Map<String, String> properties) {
-        String apiContext = properties.remove(Constants.API_CONTEXT);
-        properties.remove(Constants.USER_NAME);
-        eventMap.put(Constants.API_CONTEXT, apiContext);
-        eventMap.put(Constants.PROPERTIES, properties);
+
+        if (properties.get(Constants.API_CONTEXT) != null) {
+            eventMap.put(Constants.API_CONTEXT, properties.get(Constants.API_CONTEXT));
+        }
```
Comment (ruks) [inline component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/cloud/DefaultResponseMetricEventBuilder.java:125]: this changes might break moseif integration. Hence shall we revert this?

Comment (YasasRangika) [reply component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/cloud/DefaultResponseMetricEventBuilder.java:125]: @pubudu538 can we remove this?
it was there from the beginning [1]

[1] https://github.com/wso2/apim-analytics-publisher/blob/1279c5ff78d251f90845375be90632ad9c487bc4/component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/cloud/DefaultResponseMetricEventBuilder.java#L124

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/elk/ELKCounterMetric.java (hunk 1 above)
Comment (YasasRangika) [reply component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/elk/ELKCounterMetric.java]: Fixed with https://github.com/wso2/apim-analytics-publisher/pull/96/commits/b1dfbfdcbd91651aebfef43686335037cc69b5a6

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/cloud/DefaultResponseMetricEventBuilder.java (hunk 2 above)
Comment (YasasRangika) [reply component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/cloud/DefaultResponseMetricEventBuilder.java]: Fixed with https://github.com/wso2/apim-analytics-publisher/pull/96/commits/b58f780af5fb3e36c42831ab025087c5111e21d1

**Review by ruks — APPROVED** [review]

**Review by pubudu538 — APPROVED** [review]

**Pull request merged**